tt_llk_wormhole_b0/llk_lib/llk_unpack_A.h

```diff
@@ -75,10 +77,24 @@ inline void _llk_unpack_A_mop_config_(
 
     if (unpack_to_dest && is_32bit_input(unpack_src_format, unpack_dst_format))
     {
-        const uint32_t outerloop     = num_faces;
-        constexpr uint32_t innerloop = 1;
-        ckernel_template tmp(outerloop, innerloop, unpack_srca_to_dest);
-        tmp.program(instrn_buffer);
+        if (transpose_of_faces)
+        {
+            const uint32_t outerloop = num_faces < 4 ? 1 : 2;
+            const uint32_t innerloop = num_faces < 2 ? 1 : 2;
+            ckernel_template tmp(outerloop, innerloop, num_faces > 2 ? unpack_srca_to_dest_transpose_of_faces : unpack_srca_to_dest);
+            if (num_faces > 2)
+            {
+                tmp.set_end_op(TT_OP_SETADCZW(p_setadc::UNP_A, 0, 2, 0, 1, 0b0101));
+            }
+            tmp.program(instrn_buffer);
+        }
```

Comment:
I think I can simplify this as follows:

```cpp
if (transpose_of_faces && num_faces > 2)
{
    const uint32_t outerloop = 2;
    const uint32_t innerloop = 2;
    ckernel_template tmp(outerloop, innerloop, unpack_srca_to_dest_transpose_of_faces);
    tmp.set_end_op(TT_OP_SETADCZW(p_setadc::UNP_A, 0, 2, 0, 1, 0b0101));
    tmp.program(instrn_buffer);
}
```

I originally just copied the same approach as the non-32-bit transpose_of_faces code further down.  However, I don't see how this would work for 3 faces, presumably that's not supported at all?